### PR TITLE
Fix CI workflow and examples tests

### PR DESCRIPTION
**Pull request by renzobanegass**
## Summary

[Issue #1190](https://github.com/near/near-sdk-rs/issues/1190)

This PR addresses the following issues:
- Ensures examples `.wasm` files are built before running tests by updating CONTRIBUTING.md.
- Modifies CI workflow to build and test all examples in exception of factory-contracts.
- Adds missing unit-testing feature for `examples/non-fungible-token` and `examples/versioned`.
- Refactors deprecated features in examples.

## Changes

- Updated CONTRIBUTING.md to include instructions for building examples.
- Modified `.github/workflows/test_examples_small.yml` and `.github/workflows/test_examples.yml` to build and test more examples.
- Added unit-testing feature for `examples/non-fungible-token` and `examples/versioned` in their Cargo.toml.
- Replaced deprecated `UnorderedMap` with `LookupMap` in `examples/versioned`.

## Testing

All tests have been run locally and pass successfully. In exception for the factory-contract tests. Also the CI has been modified and we need to check which errors come up when the pipeline runs.

## LookupMap

Regarding this change I found that LookupMap is a suitable replacement for UnorderedMap based on the NEAR SDK documentation and the NEAR collections overview.

### Key Points from Documentation:

LookupMap Features:

- LookupMap provides efficient key-value storage with O(1) complexity for access, insertion, and deletion operations.
- It is a good alternative for use cases where the order of elements does not matter.

UnorderedMap Deprecation:

- UnorderedMap is deprecated, meaning it's no longer recommended for use in new code.
- The NEAR SDK documentation and source code changes have suggested moving towards other data structures like LookupMap or TreeMap for similar use cases.

### CODE REVIEW

**Comment by renzobanegass** [comment]
I fixed some typos and errors I introduced in the yaml files, now the pipeline fails as expected on the factory-contract, I'll check later when test small examples finishes if that's still working properly as I didn't really modify it, I reverted the changes I made. 

As for the factory-contract error I really don't know much about this so I would appreciate any help.

**Comment by renzobanegass** [comment]
Ok, I found a workaround by running status-message before factory-contract as it seems it needs it to be built before. Now the tests are running but there seems to be an error in the non-fungible-token regarding an unused trait, but this already seems a bit out of scope in regards to the original issue. I wait for any feedback on how this should be handled.

**Comment by frol** [comment]
@renzobanegass Add `pub` in front of `trait ValueReturnTrait` in `examples/non-fungible-token/test-approval-receiver/src/lib.rs` line 18, this way it will communicate to Rust compiler that it is going to be used publically and won't mark it as "not used".

**Comment by renzobanegass** [comment]
> @renzobanegass Add `pub` in front of `trait ValueReturnTrait` in `examples/non-fungible-token/test-approval-receiver/src/lib.rs` line 18, this way it will communicate to Rust compiler that it is going to be used publically and won't mark it as "not used".

I also added it to the `test-token-receiver/src-lib.rs:18` as it had the same warning when running tests locally.

**Comment by frol** [comment]
> The NEAR SDK documentation and source code changes have suggested moving towards other data structures like LookupMap or TreeMap for similar use cases.

Could you, please, contribute to those places to recommend IterableMap instead? (It got available with the most recent release, which is why it is still not widely recommended - we did not catch all the places)

**Comment by renzobanegass** [comment]
> > The NEAR SDK documentation and source code changes have suggested moving towards other data structures like LookupMap or TreeMap for similar use cases.
> 
> Could you, please, contribute to those places to recommend IterableMap instead? (It got available with the most recent release, which is why it is still not widely recommended - we did not catch all the places)

Should I change the LookupMap to an IterableMap instead?

**Comment by frol** [comment]
> Should I change the LookupMap to an IterableMap instead?

Yes, please

**Review by frol — APPROVED** [review]
@renzobanegass I'd like to also invite you to join @race-of-sloths. Just mention the bot in a GitHub comment. Here is how it looks: https://github.com/near/near-sdk-rs/pull/1230#issuecomment-2262451964

**Comment by renzobanegass** [comment]
@race-of-sloths include this PR please

**Comment by race-of-sloths** [comment]
@renzobanegass Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
New Sloth joined the Race! Welcome!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/renzobanegass?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/renzobanegass?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/renzobanegass)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/renzobanegass)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Comment by race-of-sloths** [comment]
### 🔄 The PR has been merged
@frol, please score the PR with `@race-of-sloths score [1/2/3/5/8/13]`. The contributor deserved it
As an alternative option auto-scoring [1, 2] will be applied to this PR!

**Comment by frol** [comment]
@race-of-sloths score 3

**Comment by race-of-sloths** [comment]
### 🌟 Score recorded!

@frol, thank you for scoring this pull request in the Race of Sloths!

**Pull request merged**
